File: Kattis/DiceGame/Dice.java
import java.util.*;

public class Dice {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);

    int counter = 0;

    int a1 = in.nextInt();
    int b1 = in.nextInt();
    int a2 = in.nextInt();
    int b2 = in.nextInt();
    int gunnarCount = 0;
    for (int i = a1; i <= b1; i++) {
      for (int j = a2; j <= b2; j++) {
        counter++;
        gunnarCount += i + j;
      }
    }
    double gunnar = gunnarCount / (double)counter;


    a1 = in.nextInt();
    b1 = in.nextInt();
    a2 = in.nextInt();
    b2 = in.nextInt();
    counter = 0;
    int emmaCount = 0;
    for (int i = a1; i <= b1; i++) {
      for (int j = a2; j <= b2; j++) {
        counter++;
        emmaCount += i+j;
      }
    }
    double emma = emmaCount / (double)counter;

    // System.out.println("Gunnar: " + gunnar);
    // System.out.println("Emma: " + emma);

    if (gunnar > emma) {
      System.out.println("Gunnar");
    } else if (emma > gunnar) {
      System.out.println("Emma");
    } else {
      System.out.println("Tie");
    }

  }

  private static int sumTo(int n) {
    return (n * (n+1)) / 2;
  }
}

File: ACM/DiceGame/dice.py
gunnar_dice = [int(e) for e in input().split()]
emma_dice = [int(e) for e in input().split()]

def build_list(dice):
    result = list()
    for i in range(dice[0], dice[1]+1):
        for j in range(dice[2], dice[3]+1):
            result.append(i + j)
    return result

emma, gunnar = build_list(emma_dice), build_list(gunnar_dice)
emma_count, gunnar_count = 0, 0
for e in emma:
    for g in gunnar:
        if e > g:
            emma_count += 1
        elif g > e:
            gunnar_count += 1

if emma_count > gunnar_count:
    print('Emma')
elif gunnar_count > emma_count:
    print('Gunnar')
else:
    print('Tie')
